tools/build_pytorch_libs.py

```diff
@@ -0,0 +1,283 @@
+    my_env = os.environ.copy()
+    my_env['PYTORCH_PYTHON'] = sys.executable
+    if USE_CUDNN:
+        my_env['CUDNN_LIBRARY'] = CUDNN_LIBRARY
+        my_env['CUDNN_INCLUDE_DIR'] = CUDNN_INCLUDE_DIR
+    if USE_CUDA:
+        my_env['CUDA_BIN_PATH'] = CUDA_HOME
+
+    mkdir_p(install_dir)
+    mkdir_p(build_dir)
+
+    cmake_defines(
+        cmake_args,
+        PYTHON_EXECUTABLE=sys.executable,
+        PYTHON_LIBRARY=cmake_python_library,
+        PYTHON_INCLUDE_DIR=distutils.sysconfig.get_python_inc(),
+        BUILDING_WITH_TORCH_LIBS="ON",
+        TORCH_BUILD_VERSION=version,
+        CMAKE_BUILD_TYPE=build_type,
+        BUILD_TORCH="ON",
+        BUILD_PYTHON=build_python,
+        BUILD_SHARED_LIBS=os.getenv("BUILD_SHARED_LIBS") or "ON",
+        BUILD_BINARY=check_env_flag('BUILD_BINARY'),
+        BUILD_TEST=build_test,
+        INSTALL_TEST=build_test,
+        BUILD_CAFFE2_OPS=not check_negative_env_flag('BUILD_CAFFE2_OPS'),
+        ONNX_NAMESPACE=os.getenv("ONNX_NAMESPACE") or "onnx_torch",
+        USE_CUDA=USE_CUDA,
+        USE_DISTRIBUTED=USE_DISTRIBUTED,
+        USE_FBGEMM=not (check_env_flag('NO_FBGEMM') or check_negative_env_flag('USE_FBGEMM')),
+        USE_NUMPY=USE_NUMPY,
+        NUMPY_INCLUDE_DIR=NUMPY_INCLUDE_DIR,
+        USE_SYSTEM_NCCL=USE_SYSTEM_NCCL,
+        NCCL_INCLUDE_DIR=NCCL_INCLUDE_DIR,
+        NCCL_ROOT_DIR=NCCL_ROOT_DIR,
+        NCCL_SYSTEM_LIB=NCCL_SYSTEM_LIB,
+        CAFFE2_STATIC_LINK_CUDA=check_env_flag('USE_CUDA_STATIC_LINK'),
+        USE_ROCM=USE_ROCM,
+        USE_NNPACK=USE_NNPACK,
+        USE_LEVELDB=check_env_flag('USE_LEVELDB'),
+        USE_LMDB=check_env_flag('USE_LMDB'),
+        USE_OPENCV=check_env_flag('USE_OPENCV'),
+        USE_QNNPACK=USE_QNNPACK,
+        USE_TENSORRT=check_env_flag('USE_TENSORRT'),
+        USE_FFMPEG=check_env_flag('USE_FFMPEG'),
+        USE_SYSTEM_EIGEN_INSTALL="OFF",
+        USE_MKLDNN=USE_MKLDNN,
+        NCCL_EXTERNAL=USE_CUDA,
+        CMAKE_INSTALL_PREFIX=install_dir,
+        CMAKE_C_FLAGS=cflags,
+        CMAKE_CXX_FLAGS=cflags,
+        CMAKE_EXE_LINKER_FLAGS=ldflags,
+        CMAKE_SHARED_LINKER_FLAGS=ldflags,
+        THD_SO_VERSION="1",
+        CMAKE_PREFIX_PATH=os.getenv('CMAKE_PREFIX_PATH') or distutils.sysconfig.get_python_lib(),
+        BLAS=os.getenv('BLAS'),
+        CUDA_NVCC_EXECUTABLE=os.getenv('CUDA_NVCC_EXECUTABLE'),
+        USE_REDIS=os.getenv('USE_REDIS'),
+        USE_GLOG=os.getenv('USE_GLOG'),
+        USE_GFLAGS=os.getenv('USE_GFLAGS'))
+
+    if USE_GLOO_IBVERBS:
+        cmake_defines(cmake_args, USE_IBVERBS="1", USE_GLOO_IBVERBS="1")
+
+    expected_wrapper = '/usr/local/opt/ccache/libexec'
+    if IS_DARWIN and os.path.exists(expected_wrapper):
+        cmake_defines(cmake_args,
+                      CMAKE_C_COMPILER="{}/gcc".format(expected_wrapper),
+                      CMAKE_CXX_COMPILER="{}/g++".format(expected_wrapper))
+    pprint(cmake_args)
+    check_call(['printenv'])
```

Comment:
This command will cause an FileNotFound error on Windows. We could replace this with `set` on Windows.

Reply:
It doesn't fail in CI because we are using Git Bash here.